src/utils/venue.ts

```diff
@@ -140,3 +140,34 @@ export const withVenue = <T extends object>(
   ...obj,
   venue,
 });
+
+export enum VenueSortingOptions {
+  az = "A - Z",
+  za = "Z - A",
+  newestFirst = "Newest First",
+  oldestFirst = "Oldest First",
+}
+
+export const sortVenues = (
+  venueList: WithId<AnyVenue>[],
+  sortingOption: VenueSortingOptions
+) => {
+  switch (sortingOption) {
+    case VenueSortingOptions.az:
+      return [...venueList].sort((a, b) => a.id.localeCompare(b.id));
+    case VenueSortingOptions.za:
+      return [...venueList].sort((a, b) => -1 * a.id.localeCompare(b.id));
+    case VenueSortingOptions.oldestFirst:
+      return [...venueList].sort(
+        (a, b) =>
+          (a.createdAt ?? 0) - (b.createdAt ?? 0) || a.id.localeCompare(b.id)
+      );
+    case VenueSortingOptions.newestFirst:
+      return [...venueList].sort(
+        (a, b) =>
+          (b.createdAt ?? 0) - (a.createdAt ?? 0) || a.id.localeCompare(b.id)
+      );
+    default:
+      return venueList;
```

Comment:
Maybe it's better to `console.error` or `throw Error` to indicate an invalid situation.

Reply:
i don't think we should do that. I guess we just can extend the VenueSortingOptions enum but forget to update the function or add a "default"/"no sorting" option for example which does nothing with the list.

Reply:
It's just a safe approach: if a new sorting option is added the function will stop working _silently_. Some sort of indication of an invalid behavior is needed IMO. Maybe the best option is to use can use an approach with `never` described [here](https://dev.to/ddiprose/exhaustive-switch-statement-with-typescript-26dh). A function like this is created:
```typescript
function assertUnreachable(x: never): never {
  throw new Error("Didn't expect to get here");
}
```
and you use that function in the `default` case. Then TS will not compile the code if the switch over the enum is not exhaustive. You can check the example behaviour of this on this [TS playground](https://www.typescriptlang.org/play?#code/KYOwrgtgBAsgngUXNA3gKClAwgQwM7ACMUAvFAES4GHkA0G2+wATKRVS3QxwMxuVMe5NAF80aAMYB7EHgAuUAJYgADmDkAuWImRt4SSADoOxfFBwg4AbnEAzMCAlzFM83gIAnOQFUQH4DgSABY4AEYANsAAFAAeWiDAAG7AHgCU8UkpUOiYckEeUgDuUAnFCB4FHlHkACKKACYgAOQKwDEqwE5QclJQAObACkEpwOSpNmJoeIWKcsFRympyqdkMEkzaBhDGTIQaDJjSslKRhuFSfVFNKgUSwO5Q6wRQhACETeMHUKH+OADWNkOG30yB2BGY+0whxkeBOwDOFyuNykdweT2AUGY70+UO+vwBa2BOiMvEhUKOsNO50u11u9zwjw2WI+gKhPwCBMw9WAthwYHCmi++E8Pj8AWCYUiC1U6k+IiAA): if you remove the `case MyEnum.Case3:` from switch TS shows an error.

Reply:
good point. 🙃 thank you!
i've added the function to a separate utils file so that it could be used in other places too. check here: https://github.com/sparkletown/sparkle/pull/2037/commits/44ddb8a8f1bb94c7cec345783953888672355814

Reply:
I dug into it a little bit more and found out that this approach works because `assertUnreachable` must have a parameter of type `never` but not the return value. Return value was specified in the article for the sake of completeness I guess.
You can check that it doesn't work at [this](https://www.typescriptlang.org/play?#code/KYOwrgtgBAsgngUXNA3gKClAwgQwM7ACMUAvFAES4GHkA0G2+wATKRVS3QxwMxuVMe5NAF80aAMYB7EHgAuUAJYgADmDkAuWImRt4SSADoOxfFBwg4AbnEAzMCAlzFM83gIAnOQFUQH4DgSABY4AEYANsAAFACUWiDAAG7AHlDomHJBHlIA7lAJeQge2R5R5AAiigAmIADkCsAAHirATlByUlAA5sAKQSnA5DE2Ymh4OYpywVHKanIxaQwSTNoGEMZMhBoMmNKyUpGG4VJdUbUq2RLA7lDLBFCEAIS1wztQof44ANY2uyv6yA2BGY20wuxkeAOwCOJzOFykVxud2AUGYz1eYPenx+DCqwFsODA4U0b3wnh8fgCwTCkViIyAA) TS playground: I removed the `MyEnum.Case3` from the switch and it still compiles.

So the way it works is like that: 
- when you exhaust all the values of an enum in a switch the compiler knows that flow will never reach a `default` clause. 
- because of that it infers the type of variable under switch statement to be `never`
- which matches the type of a parameter of the `assertUnreachable` function

[Docs](https://www.typescriptlang.org/docs/handbook/basic-types.html#never) on `never` type (I definitely learned something new🙂).

Reply:
🙃 haha me too! Thanks for the deep digging.
fixed here: https://github.com/sparkletown/sparkle/pull/2037/commits/b23ae046845056dd17f7c8f01318f5003aae4cfd